src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -1,32 +1,90 @@
 
     private final OverseasEntitySubmissionsRepository overseasEntitySubmissionsRepository;
+    private final TransactionService transactionService;
 
     private final OverseasEntityDtoDaoMapper overseasEntityDtoDaoMapper;
 
     @Autowired
     public OverseasEntitiesService(OverseasEntitySubmissionsRepository overseasEntitySubmissionsRepository,
+                                   TransactionService transactionService,
                                    OverseasEntityDtoDaoMapper overseasEntityDtoDaoMapper) {
         this.overseasEntitySubmissionsRepository = overseasEntitySubmissionsRepository;
+        this.transactionService = transactionService;
         this.overseasEntityDtoDaoMapper = overseasEntityDtoDaoMapper;
     }
 
-    public void createOverseasEntity(OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
-        ApiLogger.debug("Called createOverseasEntity()");
+    public ResponseEntity<Object> createOverseasEntity(Transaction transaction,
+                                                       OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+                                                       String passthroughTokenHeader) throws ServiceException {
+        ApiLogger.debug("Called createOverseasEntity(...)");
+
+        if (hasExistingOverseasEntitySubmission(transaction)) {
+            return ResponseEntity.badRequest().body(String.format("Transaction id: %s has an existing Overseas Entity submission", transaction.getId()));
+        }
 
+        // add the overseas entity submission into MongoDB
         var overseasEntitySubmissionDao = overseasEntityDtoDaoMapper.dtoToDao(overseasEntitySubmissionDto);
-        var insertedSubmissionDao = overseasEntitySubmissionsRepository.insert(overseasEntitySubmissionDao);
+        var insertedSubmission = overseasEntitySubmissionsRepository.insert(overseasEntitySubmissionDao);
+        var submissionUri = String.format("/transactions/%s/overseas-entity/%s", transaction.getId(), insertedSubmission.getId());
+        insertedSubmission.setLinks(Collections.singletonMap("self", submissionUri));
+        overseasEntitySubmissionsRepository.save(insertedSubmission);
+
+        // add a link to our newly created Overseas Entity submission (aka resource) to the transaction
+        var overseasEntityResource = createOverseasEntityTransactionResource(submissionUri);
+        addOverseasEntityResourceToTransaction(transaction, passthroughTokenHeader, submissionUri, overseasEntityResource);
+
+        ApiLogger.info(String.format("Overseas Entity Submission created for transaction id: %s with overseas-entity id: %s",  transaction.getId(), insertedSubmission.getId()));
+        var overseasEntitySubmissionCreatedResponseDto = new OverseasEntitySubmissionCreatedResponseDto();
+        overseasEntitySubmissionCreatedResponseDto.setId(insertedSubmission.getId());
+        return ResponseEntity.created(URI.create(submissionUri)).body(overseasEntitySubmissionCreatedResponseDto);
```

Comment:
Not an issue but just curious what the `URI.create(submissionUri)` does/achieves?

Reply:
ResponseEntity.created requires the URI as a parameter, that uri is then returned in the response location header and allows the client to access the resource if they wish